### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by sagararyal**
Fixes #

PLEASE ADD TODO LIST HERE!

Open issues:
- [x] Crash opening donate button on TPO page: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/341#issuecomment-719529615
- [x] Crash opening TPO page without `userInfo` value in localStorage: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/341#issuecomment-719704538
- [x] Edit user profile layout is not showing correctly for mobile devices: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/341#issuecomment-719906952
- [x] Crash on logout: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/341#issuecomment-719909932
- [x] Error after new user registration: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/341#issuecomment-720071178    Solved by changing config in Auth0 (@sagararyal  please confirm)
- [ ] Issues adding a new project: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/341#issuecomment-721116249
- [x] Issues editing a project: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/341#issuecomment-721122334

### CODE REVIEW

File: src/features/public/PublicUserProfile/components/ProjectSnippet.tsx [hunk 1]
```diff
@@ -1,90 +0,0 @@
...
-import i18next from '../../../../../i18n';
-import getFormatedCurrency from '../../../../utils/countryCurrency/getFormattedCurrency';
-
-const { useTranslation } = i18next;
-interface Props {
-  project: any;
-  key: number;
-}
-
-export default function ProjectSnippet({ project, key }: Props): ReactElement {
```
Comment (norbertschuler) [inline src/features/public/PublicUserProfile/components/ProjectSnippet.tsx:14]: I just put quite some effort in makes the PR https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/287 work which is doing some changes in this file - so where is this code going to? How is the TPO page done now?

Comment (harshvitra) [reply src/features/public/PublicUserProfile/components/ProjectSnippet.tsx:14]: We are using the same component used for showing the home page of the app, thanks to the new API format

**Comment by norbertschuler** [comment]
Donate button on TPO page currently is throwing these two errors:
```
Unhandled Runtime Error
TypeError: project.taxDeductionCountries.includes is not a function

Source
src/features/public/Donations/components/TreeDonation.tsx (71:38) @ eval

  69 | 
  70 |   React.useEffect(() => {
> 71 |     if (project.taxDeductionCountries.includes(country)) {
     |                                      ^
  72 |       setIsTaxDeductible(true);
  73 |     } else {
  74 |       setIsTaxDeductible(false);
```

```
Unhandled Runtime Error
TypeError: taxDeductionCountries.forEach is not a function

Source
src/features/public/Donations/components/treeDonation/SelectTaxDeductionCountryModal.tsx (56:26) @ eval

  54 | React.useEffect(() => {
  55 |   const tempCountriesData: any = [];
> 56 |   taxDeductionCountries.forEach((countryCode: string) =>
     |                        ^
  57 |     tempCountriesData.push(getCountryDataBy('countryCode', countryCode))
  58 |   );
  59 |   setCountriesData(tempCountriesData);
```

**Comment by norbertschuler** [comment]
- [ ] The layout of the edit profile page does not work on smaller screen, e.g. on my iPhone X:

IMG_8556

**Comment by norbertschuler** [comment]
- [ ] I get a 400 bad request when trying to use logout as following: 
```
GET https://accounts.plant-for-the-planet.org/v2/logout?client_id=8EyITuZvf63r8J8mjuYoaJ5Z1GJVm3zc&returnTo=auth0.pp.eco 400 (Bad Request)
```
Bildschirmfoto 2020-10-31 um 10 25 02

**Comment by norbertschuler** [comment]
- [ ] After registration of a new user on auth0.pp.eco I just got an error `Error auth0.pp.eco` in the browser showing the URL https://auth0.pp.eco/api/auth/error?error=oAuthCallback besides I got sent a confirmation mail for the registration.

**Comment by sagararyal** [comment]
Login Edge Cases


When a user verifies the email.

https://auth0.pp.eco/?supportSignUp=true&supportForgotPassword=true&email=sagar%2B1%40aryal.me&message=Your%20email%20was%20verified.%20You%20can%20continue%20using%20the%20application.&success=true&code=success

And when user is redirected with Unverified Email.
Screen Shot 2020-11-01 at 9 51 03 PM

This is being called because of a rule I set in Auth0.


Let's discuss how to deal with them properly.

We could also allow user to complete signup and ask them to confirm email afterwards.

**Comment by norbertschuler** [comment]
- [ ] This is probably debugging code? 

Bildschirmfoto 2020-11-03 um 14 31 24

**Comment by norbertschuler** [comment]
I got the following issues editing a project:

- [ ] When editing the basic data, the cost per tree input field is not shown. Only after uncheck and check the receive donations checkbox the field is shown again:

Bildschirmfoto 2020-11-03 um 14 36 12
Bildschirmfoto 2020-11-03 um 14 36 30

- [ ] I am allowed to enter an amount larger then the highest amount allowed, e.g. 10€ here:

Bildschirmfoto 2020-11-03 um 14 41 00

**Comment by norbertschuler** [comment]
I have no clue yet to which CDN the development backend might have uploaded my assets, but https://cdn.plant-for-the-planet.org/media/cache/project/medium/5fa158cb4fb02857645506.png or https://cdn.planetapp.workers.dev/media/cache/project/medium/5fa158cb4fb02857645506.png or https://cdn.pp.eco/media/cache/project/medium/5fa158cb4fb02857645506.png are not allowed for the uploaded project image:

Bildschirmfoto 2020-11-03 um 14 52 37

@sagararyal do you have an overview documentation of the different CDN endpoint and which one to use for which backend?

**Comment by harshvitra** [comment]
> * [ ]  I was not able to use a button to add some polygon for the site:

Can you try again @norbertschuler

**Review by harshvitra — APPROVED** [review]

**Pull request merged**